Libro/database

GuoqingLibro opened the pull request:
The factory for celebrities table and user table have been done, I used the 'faker' to randomly generate the data.

GuoqingLibro committed:
database settings

settings for celebrities, user, movie_celebrities

GuoqingLibro committed:
Merge branch 'main' into Libro/database

GuoqingLibro committed:
factory for celebrities

vrodriguero commented on app/Http/Controllers/User.php:
```diff
@@ -0,0 +1,10 @@
+
+use Illuminate\Http\Request;
+
+class User extends Controller
```
Hey @GuoqingLibro,

So Laravel has some rules regarding naming conventions and we also have some code formatting guidelines that I will teach during the feedbacks and reviews on the PRs. 

The Laravel convention for Controller is to have the model name accompanied by Controller on the name. So in this case, the correct format would be `UserController`

vrodriguero commented on app/Http/Controllers/celebrities.php:
```diff
@@ -0,0 +1,10 @@
+
+use Illuminate\Http\Request;
+
+class celebrities extends Controller
```
Same here ;D 
And for models and controllers, the convention is to be in the singular form. Therefore, CelebrityController is the correct format.

If you run the command 
`php artisan make:model Celebrity -a` the classes will be created according to the convention. 
But for now just rename the necessary files and leave the command for future models ;D

vrodriguero commented on app/Http/Controllers/movie_celebrities.php:
```diff
@@ -0,0 +1,10 @@
+
+use Illuminate\Http\Request;
+
+class movie_celebrities extends Controller
```
Same for here, 

The correct form should be CelebrityMovieController, when it is relation Models we usually have them ordered in alphabetical order.

vrodriguero commented on app/Models/celebrities.php:
```diff
@@ -0,0 +1,22 @@
+use Illuminate\Notifications\Notifiable;
+use Illuminate\Database\Eloquent\SoftDeletes;
+
+class celebrities extends Authenticatable
```
All classes and controllers have the upper camel case format, which means that the names will always start with uppercase. So the correct would be Celebrity and it should extend Model and not Authenticatable.

vrodriguero commented on app/Models/movie_celebrities.php:
```diff
@@ -0,0 +1,11 @@
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class movie_celebrities extends Model
```
For the relations Model we usually have them ordered alphabetically, so in this case, the correct name should be CelebrityMovie

vrodriguero commented on database/migrations/2014_10_12_000000_create_users_table.php:
```diff
@@ -17,10 +17,12 @@ public function up()
             $table->string('name');
             $table->string('email')->unique();
-            $table->timestamp('email_verified_at')->nullable();
+            $table->datetime('email_verified_at');
```
Oh! This one can be a timestamp as before ;D and also nullable(), which means that the value can be null.

vrodriguero commented on database/migrations/2014_10_12_000000_create_users_table.php:
```diff
@@ -17,10 +17,12 @@ public function up()
+            $table->datetime('email_verified_at');
             $table->string('password');
-            $table->rememberToken();
+            $table->string('remember_token' );
```
Oh! This could also be the same as before, sorry I know I said string but there is actually a type for the `rememberToken();`

vrodriguero replied:
I think you missed this one 😬 , you can revert this to be as it was before: 
-> `datetime` -> `timestamp`
-> `string(remember_token)` -> `rememberToken()`

vrodriguero replied:
And this one too 😅

vrodriguero commented on database/migrations/2021_08_26_003440_create_celebrities_table.php:
```diff
@@ -0,0 +1,36 @@
+            $table->id();
+            $table->string('name');
+            $table->date('DOB');
+            $table->string('Nationality');
```
For column names usually, we leave it all lowercase. So any of the names you have that start with uppercase should be updated to lowercase. Once you changed this one you will probably have to update the fillable property on the model and the factory

vrodriguero commented on database/migrations/2021_08_26_004523_create_movie_celebrities_table.php:
```diff
@@ -0,0 +1,34 @@
+    {
+        Schema::create('movie_celebrities', function (Blueprint $table) {
+            $table->id();
+            $table->integer('movieID');
```
the convention for Laravel for foreign keys is to have the `_` so for example in this case should be `movie_id` and `celebrity_id`. This way Laravel will be able to fetch the correct model without needing much setup 😄

vrodriguero commented on database/seeders/DatabaseSeeder.php:
```diff
@@ -13,6 +13,7 @@ class DatabaseSeeder extends Seeder
     {
-        // \App\Models\User::factory(10)->create();
+        \App\Models\User::factory(10)->create();
+        \App\Models\celebrities::factory(10)->create();
```
Once you change the Celebrity model than you can update the reference here too 😄

vrodriguero commented on app/Models/User.php:
```diff
@@ -6,10 +6,11 @@
 use Illuminate\Database\Eloquent\Factories\HasFactory;
 use Illuminate\Foundation\Auth\User as Authenticatable;
 use Illuminate\Notifications\Notifiable;
+use Illuminate\Database\Eloquent\SoftDeletes;
```
For imports, we usually for the code to be ordered based on the length. So if you have any files that uses imports, it makes the code cleaner to order them in the same way ;D 

```
use App\Models\User;
use Illuminate\Support\Str;
use Illuminate\Database\Eloquent\Factories\Factory;
```

vrodriguero commented on app/Models/celebrities.php:
```diff
@@ -0,0 +1,22 @@
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Foundation\Auth\User as Authenticatable;
+use Illuminate\Notifications\Notifiable;
+use Illuminate\Database\Eloquent\SoftDeletes;
```
Only the ordering ;D

vrodriguero commented on app/Models/movie_celebrities.php:
```diff
@@ -0,0 +1,11 @@
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
```
ordering ;D

vrodriguero commented on database/factories/celebritiesFactory.php:
```diff
@@ -0,0 +1,31 @@
+namespace Database\Factories;
+
+use App\Models\celebrities;
+use Illuminate\Database\Eloquent\Factories\Factory;
```
Order Correct! 🥳 
Only change is related to the already commented Celebrity model 😄

vrodriguero reviewed (changes requested):
Hey @GuoqingLibro,

Great effort! I know that the beginning is a bit challenging but I promise things to get easier with time and practice. 

I left some comments and explained some of the guidelines that Laravel has regarding naming conventions and some code formatting tips.

GuoqingLibro committed:
coding convention changes

Based on the required changes, the codes have been updated.

vrodriguero commented on app/Http/Controllers/CelebrityMovieController.php:
```diff
@@ -0,0 +1,10 @@
+
+use Illuminate\Http\Request;
+
+class Celebrity_movie extends Controller
```
Hey @GuoqingLibro the convention for Laravel is to not have _ in the names and Controllers should include Controller in the name so the correct format should be CelebrityMovieController

vrodriguero commented on app/Http/Controllers/UserController.php:
```diff
@@ -0,0 +1,10 @@
+
+use Illuminate\Http\Request;
+
+class User extends Controller
```
Same as the comment above 😄

vrodriguero commented on app/Models/Celebrity.php:
```diff
@@ -0,0 +1,22 @@
+
+class Celebrity extends Model
+{
+    use HasFactory, Notifiable,SoftDeletes;
```
Hmm... I am not sure if we need Notifiable on the Celebrity model, as we are not planning to send notifications to a celebrity I think we are safe to remove it completely.

vrodriguero commented on app/Models/User.php:
```diff
@@ -2,14 +2,15 @@
 class User extends Authenticatable
 {
-    use HasFactory, Notifiable;
+    use HasFactory, Notifiable,SoftDeletes;
```
Just some formatting tips: it makes it more clean and consistent if we add spaces after the commas 😄 . So you can add a space after the `Notifiable,`

vrodriguero replied:
@GuoqingLibro I think we missed this one hehe

vrodriguero commented on app/Http/Controllers/CelebrityController.php:
```diff
@@ -0,0 +1,10 @@
+
+use Illuminate\Http\Request;
+
+class Celebrity extends Controller
```
Hehe, filename is correct, just need to change the class name to be the same `CelebrityController `

vrodriguero commented on database/factories/celebritiesFactory.php:
```diff
@@ -0,0 +1,31 @@
+use App\Models\Celebrity;
+use Illuminate\Database\Eloquent\Factories\Factory;
+
+class CelebrityFactory extends Factory
```
correct class name, but the filename also needs to be updated, otherwise, Laravel won't be able to find it.

vrodriguero commented on database/factories/movie_celebritiesFactory.php:
```diff
@@ -0,0 +1,28 @@
+use App\Models\CelebrityMovie;
+use Illuminate\Database\Eloquent\Factories\Factory;
+
+class CelebrityMovieFactory extends Factory
```
Same here

vrodriguero commented on database/migrations/2021_08_26_004523_create_movie_celebrities_table.php:
```diff
@@ -0,0 +1,34 @@
+use Illuminate\Database\Migrations\Migration;
+
+
+class CreateMovieCelebritiesTable extends Migration
```
This table name should reflect the model and the controller, so if the model is celebritymodel then the migration should be create_celebrity_movies_table. 

The easiest fix is to create the migration again to make sure that the name is the same as the Model and Controller

vrodriguero reviewed (commented):
From the changes on this branch, I noticed that somehow the file named Kernel was moved from the App/Http to Http/Middleware. I think that this will break things so it is best to revert this. So move back to the App/Http directory.

![Screen Shot 2021-09-02 at 12 06 26 pm](https://user-images.githubusercontent.com/53244381/131769752-c2d05d47-b4f6-4438-901b-f8da6ac8ff3a.png)

GuoqingLibro committed:
updated factory and database

The convention error has been fixed.
The factory is working now with generating faker data in user and celebrity.
The celebtity_movies relationship table is done, the data can be captured from celebrity table. But still waiting for the movie table.

vrodriguero reviewed (changes requested):
Hey Libro awesome work ;D 

Just two missed comments hehe :D 

Once you finish that I will do a final tests and I will merge this to the main ;D

GuoqingLibro committed:
Two missing comment updated

One missing comment in user model file has been fixed.
one missing comment in migration table(user) has been fixed.

vrodriguero committed:
code formatting

vrodriguero committed:
Merge branch 'main' into Libro/database

# Conflicts:
#	database/seeders/DatabaseSeeder.php

vrodriguero committed:
code formatting

vrodriguero reviewed (approved):
Nice work @GuoqingLibro !!

I just did a few last tweaks regarding the migration for celebrity_movies table, usually when we have a foreign id we use unsignedInteger as type ;D 

And I also merged the main in here and resolved the conflicts 😄
